From ff73d9e35717237ea1ace999124291fa7d708810 Mon Sep 17 00:00:00 2001
From: Lukasz Smurawski <lukasz.smurawski@grinn-global.com>
Date: Mon, 20 Apr 2026 17:56:26 +0200
Subject: [PATCH 2/2] [MCU]: Remove wifibt related code

The Astra 1680 SBC has Wi-Fi/BT-related pins connected to non-SM_* GPIOs.
Additionally, the Wi-Fi/BT code is independent of the .config,
so this patch removes it entirely.
---
 bootloader/sm_cm3/syna/Makefile               |   1 -
 bootloader/sm_cm3/syna/inc/sm_power.h         |   2 -
 .../sm_cm3/syna/src/power/sm_power_eth.c      | 284 +-----------------
 .../sm_cm3/syna/src/power/sm_power_flow.c     |   1 -
 bootloader/sm_cm3/syna/src/sm_wifibt.c        | 102 -------
 5 files changed, 2 insertions(+), 388 deletions(-)
 delete mode 100644 bootloader/sm_cm3/syna/src/sm_wifibt.c

diff --git a/bootloader/sm_cm3/syna/Makefile b/bootloader/sm_cm3/syna/Makefile
index 11bd77a..f425354 100644
--- a/bootloader/sm_cm3/syna/Makefile
+++ b/bootloader/sm_cm3/syna/Makefile
@@ -141,7 +141,6 @@ cSRCS += \
 	 $(SYNA_DIR)/src/sm_ddr_ctrl.c \
 	 $(SYNA_DIR)/src/sm_temperature.c \
 	 $(SYNA_DIR)/src/sm_led.c \
-	 $(SYNA_DIR)/src/sm_wifibt.c \
 	 $(SYNA_DIR)/src/sm_apb_timer.c \
 	 $(SYNA_DIR)/src/unit_test/gpio_test.c \
 	 $(SYNA_DIR)/src/power/sm_power_flow.c \
diff --git a/bootloader/sm_cm3/syna/inc/sm_power.h b/bootloader/sm_cm3/syna/inc/sm_power.h
index acc2088..40950d8 100644
--- a/bootloader/sm_cm3/syna/inc/sm_power.h
+++ b/bootloader/sm_cm3/syna/inc/sm_power.h
@@ -112,7 +112,6 @@ int sm_setup_power_down_timer(int msg);
 
 void sm_set_reboot_reason(UINT32 reboot_reason);
 
-void sm_wifi_reset(void);
 void eth_wifibt_wakeup_handler(void);
 void fpbutton_wakeup_handler(void);
 
@@ -129,7 +128,6 @@ hresult mv_sm_power_enterflow_bysmstate(void);
 hresult mv_sm_power_setwakeupsource(INT32 ws);
 void mv_sm_power_setwakeuppayload(unsigned int value);
 
-extern volatile BOOL is_wakeup;
 #ifdef GETHENABLE
 extern volatile BOOL geth_wakeup;
 #endif
diff --git a/bootloader/sm_cm3/syna/src/power/sm_power_eth.c b/bootloader/sm_cm3/syna/src/power/sm_power_eth.c
index d564c3d..401b7ef 100644
--- a/bootloader/sm_cm3/syna/src/power/sm_power_eth.c
+++ b/bootloader/sm_cm3/syna/src/power/sm_power_eth.c
@@ -31,299 +31,25 @@
  * DAMAGES OR ANY OTHER DAMAGES, SYNAPTICS' TOTAL CUMULATIVE LIABILITY
  * TO ANY PARTY SHALL NOT EXCEED ONE HUNDRED U.S. DOLLARS.
  */
-#include "FreeRTOS.h"
-#include "task.h"
-#include "timers.h"
-#include "semphr.h"
-
 #include "platform_config.h"
 
 #include "sm_type.h"
-#include "SysMgr.h"
 #include "sm_common.h"
 
 #include "sm_printf.h"
 #include "sm_gpio.h"
 #include "sm_state.h"
-#include "sm_timer.h"
 
-#include "sm_rt_module.h"
 #include "sm_power.h"
-#include "sm_func.h"
-#include "sm_mem_map_itcm.h"
-#include "sm_io.h"
-#include "sm_memmap.h"
-
-#ifndef WIFIBT_GAP_MIN
-#define WIFIBT_GAP_MIN	10
-#define WIFIBT_GAP_MAX	125
-#define ETH_GAP_MAX	512
-#endif
-
-/* define the specific gap in config if different */
-#ifndef BTKEY_NETFLIX_GAP_MIN
-#define BTKEY_NETFLIX_GAP_MIN 395
-#define BTKEY_NETFLIX_GAP_MAX 415
-#endif
-
-#define WIFIBT_STACK_SIZE  ((uint16_t) 128)
-
-int wifibt_falling_time = 0;
-int wifibt_gap = 0;
-
-static TimerHandle_t wifibt_timer = NULL;
-static SemaphoreHandle_t wifibt_sem = NULL;
-
-volatile BOOL is_wakeup = FALSE;
-typedef struct {
-	unsigned int times;
-	unsigned int level;
-	BOOL is_btinfo;
-}Itcm_bt_info;
-
-static void sm_trigger_irq(void)
-{
-	unsigned int reg;
-
-	reg = MV_SM_READ_REG32(SM_MEMMAP_SYS_CTRL_BASE + RA_smSysCtl_SM_CTRL);
-	reg |= 0x2; //enable sw interrupt
-	MV_SM_WRITE_REG32(SM_MEMMAP_SYS_CTRL_BASE + RA_smSysCtl_SM_CTRL, reg);
-}
-
-static TimerHandle_t wifireset_timer = NULL;
-static Itcm_bt_info *g_itcm = (Itcm_bt_info *)(uintptr_t)(SM_BT_INFO_BASE);
-void sm_gpio_int_pre(int port)
-{
-	sm_gpio_int_disable(port);
-	sm_gpio_clearint(port);
-	sm_gpio_set_input(port);
-
-	g_itcm->times = 0;
-	g_itcm->level = 0;
-	g_itcm->is_btinfo = FALSE;
-}
-
-BOOL state_to_wait_wakeup(void)
-{
-	MV_SM_STATE cur_state = mv_sm_get_state();
-
-	if((MV_SM_STATE_LOWPOWERSTANDBY == cur_state) ||
-		(MV_SM_STATE_NORMALSTANDBY == cur_state) ||
-		(MV_SM_STATE_SUSPEND == cur_state)) {
-		return TRUE;
-	}
-
-	return FALSE;
-}
-
-static void sm_wifireset_timer_isr(TimerHandle_t xtimer)
-{
-	board_wifi_poweron();
-
-	if(xTimerStop(wifireset_timer, 0) == pdFAIL)
-		PRT_ERROR("wifireset timer can't be stopped\n");
-}
-
-void sm_wifi_reset(void)
-{
-	board_wifi_poweroff();
-
-	if (wifireset_timer != NULL) {
-		if ( xTimerIsTimerActive(wifireset_timer) == pdFALSE ) {
-			if(xTimerStart(wifireset_timer, pdMS_TO_TICKS(1)) == pdFAIL)
-				PRT_ERROR("wifireset timer can't be started\n");
-		}
-	}
-}
-
-static void sm_wifibt_timer_isr(TimerHandle_t xtimer)
-{
-	wifibt_gap = mv_sm_timer_gettimems_formISR() - wifibt_falling_time;
-
-#ifdef SM_GPIO_PORT_WIFIBT
-	if (sm_gpio_read(SM_GPIO_PORT_WIFIBT) == RISING_EDGE) {
-		is_wakeup = TRUE;
-
-		PRT_INFO("wifibt_gap: %d\n", wifibt_gap);
-		if(wifibt_gap > (WIFIBT_GAP_MIN * 2))
-			mv_sm_power_setwakeupsource(MV_SM_WAKEUP_SOURCE_BT);
-		else
-			mv_sm_power_setwakeupsource(MV_SM_WAKEUP_SOURCE_WIFI);
-	}
-#else
-	if (sm_gpio_read(SM_GPIO_PORT_BT) == RISING_EDGE) {
-		is_wakeup = TRUE;
-		PRT_INFO("wifibt_gap: %d\n", wifibt_gap);
-		mv_sm_power_setwakeupsource(MV_SM_WAKEUP_SOURCE_BT);
-	}
-#endif
-
-	if (is_wakeup || wifibt_gap > WIFIBT_GAP_MAX*10) {
-		/* cover either separate bt wakeup pin or combined wifibt wakeup pin */
-		if ((wifibt_gap > BTKEY_NETFLIX_GAP_MIN) &&
-			(wifibt_gap < BTKEY_NETFLIX_GAP_MAX))
-			mv_sm_power_setwakeuppayload(KEY_NETFLIX);
-		else if (wifibt_gap > (WIFIBT_GAP_MIN * 2))
-			mv_sm_power_setwakeuppayload(KEY_OTHERS);
-
-		if(xTimerStop(wifibt_timer, 0) == pdFAIL)
-			PRT_ERROR("wifibt timer can't be stopped\n");
-		wifibt_gap = 0;
-	}
-}
-
-#ifdef SM_GPIO_PORT_WIFIBT
-static void __attribute__((used)) sm_wifibt_isr(void)
-{
-	sm_gpio_clearint(SM_GPIO_PORT_WIFIBT);
-
-	g_itcm->times ++;
-	g_itcm->is_btinfo = TRUE;
-	if (sm_gpio_read(SM_GPIO_PORT_WIFIBT) == RISING_EDGE)
-		g_itcm->level = 1;
-	else
-		g_itcm->level = 0;
-
-	if(wifibt_gap == 0)
-		wifibt_falling_time = mv_sm_timer_gettimems_formISR();
-
-	if ( wifibt_sem != NULL ) {
-		BaseType_t xHigherPriorityTaskWoken = pdFALSE;
-		xSemaphoreGiveFromISR(wifibt_sem, &xHigherPriorityTaskWoken);
-	}
-
-	sm_trigger_irq();
-}
-DECLARE_GPIO_ISR(SM_GPIO_PORT_WIFIBT, EDGE_SENSITIVE, FALLING_EDGE, sm_wifibt_isr);
-
-#else
-static void __attribute__((used)) sm_wifi_isr(void)
-{
-	sm_gpio_clearint(SM_GPIO_PORT_WIFI);
-
-	if(state_to_wait_wakeup()) {
-		is_wakeup = TRUE;
-		mv_sm_power_setwakeupsource(MV_SM_WAKEUP_SOURCE_WIFI);
-	}
-}
-DECLARE_GPIO_ISR(SM_GPIO_PORT_WIFI, EDGE_SENSITIVE, FALLING_EDGE, sm_wifi_isr);
-
-static void __attribute__((used)) sm_bt_isr(void)
-{
-	sm_gpio_clearint(SM_GPIO_PORT_BT);
-
-	g_itcm->times ++;
-	g_itcm->is_btinfo = TRUE;
-	if (sm_gpio_read(SM_GPIO_PORT_BT) == RISING_EDGE)
-		g_itcm->level = 1;
-	else
-		g_itcm->level = 0;
-
-	if(wifibt_gap == 0)
-		wifibt_falling_time = mv_sm_timer_gettimems_formISR();
-
-	if ( wifibt_sem != NULL ) {
-		BaseType_t xHigherPriorityTaskWoken = pdFALSE;
-		xSemaphoreGiveFromISR(wifibt_sem, &xHigherPriorityTaskWoken);
-	}
-
-	sm_trigger_irq();
-}
-DECLARE_GPIO_ISR(SM_GPIO_PORT_BT, EDGE_SENSITIVE, FALLING_EDGE, sm_bt_isr);
-#endif
-
-static void wifibt_wake_task(void *para)
-{
-	while (1) {
-		xSemaphoreTake(wifibt_sem, portMAX_DELAY);
-
-		if(state_to_wait_wakeup()) {
-			if (wifibt_timer != NULL) {
-				if ( xTimerIsTimerActive(wifibt_timer) == pdFALSE ) {
-					if(xTimerStart(wifibt_timer, pdMS_TO_TICKS(1)) == pdFAIL)
-						PRT_ERROR("wifibt timer can't be started\n");
-				}
-			}
-		}
-	}
-}
-
-static void __attribute__((used)) create_wifibt_wake_task(void)
-{
-	/* Create timer trigger task */
-	xTaskCreate(wifibt_wake_task, "wifibt_wake", WIFIBT_STACK_SIZE, NULL,
-		TASK_PRIORITY_2, NULL);
-}
-
-DECLARE_RT_MODULE(
-	power_eth,
-	MV_SM_ID_ETH,
-	create_wifibt_wake_task,
-	NULL,
-	NULL
-);
-
-static int __attribute__((used)) power_eth_init(void)
-{
-	/* Clean up interrupt & set WiFiBT wake pin direction */
-#ifdef SM_GPIO_PORT_WIFIBT
-	sm_gpio_int_pre(SM_GPIO_PORT_WIFIBT);
-#else
-	sm_gpio_int_pre(SM_GPIO_PORT_WIFI);
-	sm_gpio_int_pre(SM_GPIO_PORT_BT);
-#endif
-
-	/* Create notify semaphore */
-	wifibt_sem = xSemaphoreCreateBinary();
-	if (wifibt_sem == NULL)
-		PRT_ERROR("wifibt semaphore create failed\n");
-
-	/* Crate Wifibt timer */
-	if (wifibt_timer == NULL) {
-		// Create timer during system init
-		wifibt_timer = xTimerCreate("wifibt",
-				pdMS_TO_TICKS(5), // 5ms
-				pdTRUE,
-				0,
-				sm_wifibt_timer_isr);
-	}
-	if (wifibt_timer == NULL)
-		PRT_ERROR("wifibt timer request failed\n");
-
-	/* Crate Wifireset timer */
-	if (wifireset_timer == NULL) {
-		// Create timer during system init
-		wifireset_timer = xTimerCreate("wifireset",
-				pdMS_TO_TICKS(200), // 200ms
-				pdFAIL,
-				0,
-				sm_wifireset_timer_isr);
-	}
-	if (wifireset_timer == NULL)
-		PRT_ERROR("wifireset timer request failed\n");
-
-#ifdef SM_GPIO_PORT_WIFIBT
-	sm_gpio_int_enable(SM_GPIO_PORT_WIFIBT);
-#else
-	sm_gpio_int_enable(SM_GPIO_PORT_WIFI);
-	sm_gpio_int_enable(SM_GPIO_PORT_BT);
-#endif
-
-	return S_OK;
-}
-
-DECLARE_RT_INIT(power_eth, INIT_DEV_P_25, power_eth_init);
 
 #ifdef GETHENABLE
 volatile BOOL geth_wakeup = FALSE;
 
 void sm_geth_isr(void)
 {
-	if(mv_sm_get_state() != MV_SM_STATE_ACTIVE){
+	if (mv_sm_get_state() != MV_SM_STATE_ACTIVE)
 		geth_wakeup = TRUE;
-	}
 
-	// Clear GPIO interrupt
 	sm_gpio_clearint(SM_GPIO_PORT_GEPHY);
 	PRT_INFO("@@@GEPHY GPIO INT!\n");
 }
@@ -333,14 +59,8 @@ DECLARE_GPIO_ISR(SM_GPIO_PORT_GEPHY, EDGE_SENSITIVE, FALLING_EDGE, sm_geth_isr);
 
 void eth_wifibt_wakeup_handler(void)
 {
-	if(is_wakeup) {
-		mv_sm_power_enterflow_bysmstate();
-		is_wakeup = FALSE;
-	}
-
 #ifdef GETHENABLE
-	//FIXME: Keep original wifibt_gap for Ethernet wakeup temporarily until hw solution are fixed.
-	if(geth_wakeup) {
+	if (geth_wakeup) {
 		mv_sm_power_setwakeupsource(MV_SM_WAKEUP_SOURCE_WOL);
 		mv_sm_power_enterflow_bysmstate();
 		geth_wakeup = FALSE;
diff --git a/bootloader/sm_cm3/syna/src/power/sm_power_flow.c b/bootloader/sm_cm3/syna/src/power/sm_power_flow.c
index 7638e9a..7ada962 100644
--- a/bootloader/sm_cm3/syna/src/power/sm_power_flow.c
+++ b/bootloader/sm_cm3/syna/src/power/sm_power_flow.c
@@ -666,7 +666,6 @@ static int __attribute__((used)) mv_sm_process_power_msg(void * data, INT32 len)
 			{
 				is_linux_ready = TRUE;
 
-				is_wakeup = FALSE;
 				#ifdef GETHENABLE
 				geth_wakeup = FALSE;
 				#endif
diff --git a/bootloader/sm_cm3/syna/src/sm_wifibt.c b/bootloader/sm_cm3/syna/src/sm_wifibt.c
deleted file mode 100644
index 8f1374e..0000000
--- a/bootloader/sm_cm3/syna/src/sm_wifibt.c
+++ /dev/null
@@ -1,102 +0,0 @@
-/*
- * NDA AND NEED-TO-KNOW REQUIRED
- *
- * Copyright © 2013-2018 Synaptics Incorporated. All rights reserved.
- *
- * This file contains information that is proprietary to Synaptics
- * Incorporated ("Synaptics"). The holder of this file shall treat all
- * information contained herein as confidential, shall use the
- * information only for its intended purpose, and shall not duplicate,
- * disclose, or disseminate any of this information in any manner
- * unless Synaptics has otherwise provided express, written
- * permission.
- *
- * Use of the materials may require a license of intellectual property
- * from a third party or from Synaptics. This file conveys no express
- * or implied licenses to any intellectual property rights belonging
- * to Synaptics.
- *
- * INFORMATION CONTAINED IN THIS DOCUMENT IS PROVIDED "AS-IS," AND
- * SYNAPTICS EXPRESSLY DISCLAIMS ALL EXPRESS AND IMPLIED WARRANTIES,
- * INCLUDING ANY IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
- * A PARTICULAR PURPOSE, AND ANY WARRANTIES OF NON-INFRINGEMENT OF ANY
- * INTELLECTUAL PROPERTY RIGHTS. IN NO EVENT SHALL SYNAPTICS BE LIABLE
- * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, PUNITIVE, OR
- * CONSEQUENTIAL DAMAGES ARISING OUT OF OR IN CONNECTION WITH THE USE
- * OF THE INFORMATION CONTAINED IN THIS DOCUMENT, HOWEVER CAUSED AND
- * BASED ON ANY THEORY OF LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
- * NEGLIGENCE OR OTHER TORTIOUS ACTION, AND EVEN IF SYNAPTICS WAS
- * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. IF A TRIBUNAL OF
- * COMPETENT JURISDICTION DOES NOT PERMIT THE DISCLAIMER OF DIRECT
- * DAMAGES OR ANY OTHER DAMAGES, SYNAPTICS' TOTAL CUMULATIVE LIABILITY
- * TO ANY PARTY SHALL NOT EXCEED ONE HUNDRED U.S. DOLLARS.
- */
-#include "sm_type.h"
-
-#include "platform_config.h"
-
-#include "sm_common.h"
-
-#include "sm_power.h"
-#include "sm_timer.h"
-#include "sm_printf.h"
-
-#include "sm_gpio.h"
-#include "sm_rt_module.h"
-
-#ifdef SM_GPIO_WIFIBT_POWER
-
-static hresult mv_sm_wifibt_power_ctrl(BOOL on)
-{
-	if(on) {
-		sm_gpio_set_output(SM_GPIO_WIFIBT_POWER, 1);
-	} else {
-		sm_gpio_set_output(SM_GPIO_WIFIBT_POWER, 0);
-	}
-	PRT_INFO("WIFI/BT power %s, gpio: %d\n", on?"on":"off", SM_GPIO_WIFIBT_POWER);
-	return S_OK;
-}
-
-#else
-
-static hresult mv_sm_wifibt_power_ctrl(BOOL on)
-{
-	return S_OK;
-}
-
-#endif
-
-hresult mv_sm_process_internal_wifibt_msg(void * data, INT32 len)
-{
-	UINT32 *msg = (UINT32*)data;
-
-	if(msg[0] == MV_SM_ID_POWER) {
-		switch(msg[1]) {
-		case FLOW_COLD_2_LOWPOWERSTANDBY:
-		case FLOW_ACTIVE_2_LOWPOWERSTANDBY:
-			if(msg[2] == STA_LEAVEFLOW)
-				mv_sm_wifibt_power_ctrl(0);
-			break;
-		case FLOW_LOWPOWERSTANDBY_2_ACTIVE:
-			if(msg[2] == STA_ENTERFLOW)
-				mv_sm_wifibt_power_ctrl(1);
-			break;
-		case FLOW_SYSRESET_2_ACTIVE:
-			if(msg[2] == STA_ENTERFLOW)
-				sm_wifi_reset();
-			break;
-		default:
-			break;
-		}
-	}
-	return S_OK;
-}
-
-DECLARE_RT_MODULE(
-	pwifibt,
-	MV_SM_ID_WIFIBT,
-	NULL,
-	NULL,
-	mv_sm_process_internal_wifibt_msg
-);
-
-- 
2.43.0

